### PR TITLE
For #7586 - Removes mobile prefixes when toggling desktop mode

### PR DESCRIPTION
**Pull request by boek**
Tries to remove `m.` and `mobile.` from the host to re-request the desktop site with a changed user agent

### CODE REVIEW

File: Client/Frontend/Browser/Tab.swift [hunk 1]
```diff
@@ -555,10 +555,18 @@ class Tab: NSObject {
...
+            let request = URLRequest(url: url)
+            webView?.load(request)
+        } else {
+            reload()
+        }
+
         TabEvent.post(.didToggleDesktopMode, for: self)
     }
 
+
```
Comment (nbhasin2) [inline Client/Frontend/Browser/Tab.swift:569]: nit: not sure if this is extra line

**Review by nbhasin2 — APPROVED** [review]

File: Client/Frontend/Browser/Tab.swift [hunk 2]
```diff
@@ -767,3 +775,34 @@ class TabWebViewMenuHelper: UIView {
         }
     }
 }
+
+extension URL {
+
```
Comment (dnarcese) [inline Client/Frontend/Browser/Tab.swift:780]: nit: is this extra line necessary

File: Client/Frontend/Browser/Tab.swift [hunk 3]
```diff
@@ -767,3 +775,34 @@ class TabWebViewMenuHelper: UIView {
...
+    /**
+    Returns a URL without a mobile prefixe (`"m."` or `"mobile."`)
```
Comment (dnarcese) [inline Client/Frontend/Browser/Tab.swift:782]: typo "prefixe" -> "prefix"

**Comment by dnarcese** [comment]
I'm just testing it out on m.facebook.com but it isn't working. Is there a site you were testing with?
Also, what happens when we try to go back to the mobile site?

**Comment by boek** [comment]
@isabelrios any idea why the UITest is failing here? I'm also running it locally on main and its failing :( but other PRs it's passing?

**Comment by boek** [comment]
QA said tests were intermittently failing and logged an issue: https://github.com/mozilla-mobile/firefox-ios/issues/7867

This was said to be safe to land

**Comment by karlcow** [comment]
@boek This seems to have introduced a regression on shazam.com
https://github.com/webcompat/web-bugs/issues/68816

**Pull request merged**